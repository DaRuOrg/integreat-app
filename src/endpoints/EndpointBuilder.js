import { isEqual } from 'lodash/lang'

import Endpoint from './Endpoint'
import StateMapperBuilder from './StateMapperBuilder'

/**
 * Helper class to build a {@link Endpoint}
 */
class EndpointBuilder {
  _name
  _url
  _mapper
  _stateMapperBuilder
  _refetchLogic

  /**
   * Creates a new endpoint builder
   * @param {string} name The name of the endpoint to build
   */
  constructor (name) {
    this._name = name
    this._stateMapperBuilder = new StateMapperBuilder(this)

    this._refetchLogic = (urlParams, nextUrlParams) => !isEqual(urlParams, nextUrlParams)
  }

  /**
   * Adds a url to the builder
   * @param url The url
   * @return {EndpointBuilder} The builder itself
   */
  withUrl (url) {
    this._url = url
    return this
  }

  /**
   * Adds a url to the builder
   * @param mapper The mapper
   * @return {EndpointBuilder} The builder itself
   */
  withMapper (mapper) {
    this._mapper = mapper
    return this
  }

  /**
   * Adds a state mapper builder to this builder
   * @return {StateMapperBuilder} The state mapper of this builder
   */
  withStateMapper () {
    return this._stateMapperBuilder
  }

  /**
   * Adds refetch logic to this builder
   * @param refetchLogic The refetch logic
   * @return {EndpointBuilder}  The builder itself
   */
  withRefetchLogic (refetchLogic) {
    this._refetchLogic = refetchLogic
    return this
  }

  /**
   * Checks the data and builds the endpoint
   * @return {Endpoint} The final endpoint
   */
  build () {
    if (!this._name) {
      throw Error('You have to set a name to build an endpoint!')
    }

    if (!this._url) {
      throw Error('You have to set a url to build an endpoint!')
    }

    if (!this._mapper) {
      throw Error('You have to set a mapper to build an endpoint!')
    }

    if (!this._url) {
      throw Error('You have to set a url to build an endpoint!')
    }

    if (!this._stateMapperBuilder) {
      throw Error('You have to set a state mapper to build an endpoint!')
    }

    if (!this._refetchLogic) {
      throw Error('You have to set a refetch logic to build an endpoint!')
    }

    return new Endpoint(this._name, this._url, this._mapper, this._stateMapperBuilder.build(), this._refetchLogic)
  }
}

export default EndpointBuilder
