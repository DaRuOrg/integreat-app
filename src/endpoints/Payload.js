class Payload {
  constructor (isFetching, data = null, error = null, requestUrl = null) {
    this._isFetching = isFetching
    this._data = data
    this._error = error
    this._requestUrl = requestUrl
  }

  get isFetching () {
    return this._isFetching
  }

  get data () {
    return this._data
  }

  ready () {
    return !!this._data
  }

  get error () {
    return this._error
  }

  get requestUrl () {
    return this._requestUrl
  }
}

export default Payload
