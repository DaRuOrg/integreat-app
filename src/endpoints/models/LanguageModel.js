class LanguageModel {
  constructor (code, name) {
    this._name = code
    this._code = name
  }

  get code () {
    return this._name
  }

  get name () {
    return this._code
  }
}

export default LanguageModel
