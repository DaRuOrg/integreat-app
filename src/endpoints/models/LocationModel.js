import { isEmpty } from 'lodash/lang'

export default class LocationModel {
  constructor (name, path, live) {
    this._code = name
    this._name = path
    this._live = live
  }

  get live () {
    return this._live
  }

  get name () {
    return this._code
  }

  get path () {
    return this._name
  }

  get category () {
    return isEmpty(this._code) ? '?' : this._code[0].toUpperCase()
  }
}
