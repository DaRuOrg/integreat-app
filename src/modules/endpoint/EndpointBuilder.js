import { isEqual } from 'lodash/lang'

import Endpoint from './Endpoint'

/**
 * Helper class to build a {@link Endpoint}
 */
class EndpointBuilder {
  _name
  _routerToUrlMapper
  _mapper
  _responseOverride
  _refetchLogic

  /**
   * Creates a new endpoint builder
   * @param {string} name The name of the endpoint to build
   */
  constructor (name) {
    this._name = name

    this._refetchLogic = (urlParams, nextUrlParams) => !isEqual(urlParams, nextUrlParams)
  }

  /**
   * Adds a router to url mapper to the builder
   * @param routerToUrlMapper The url
   * @return {EndpointBuilder} The builder itself
   */
  withRouterToUrlMapper (routerToUrlMapper) {
    this._routerToUrlMapper = routerToUrlMapper
    return this
  }

  /**
   * Adds a url to the builder
   * @param mapper The mapper
   * @return {EndpointBuilder} The builder itself
   */
  withMapper (mapper) {
    this._mapper = mapper
    return this
  }

  /**
   * Adds refetch logic to this builder
   * @param refetchLogic The refetch logic
   * @return {EndpointBuilder} The builder itself
   */
  withRefetchLogic (refetchLogic) {
    this._refetchLogic = refetchLogic
    return this
  }

  /**
   * Overrides value from the API response. Useful for testing.
   * @param responseOverride {*} The response
   * @return {EndpointBuilder} The builder itself
   */
  withResponseOverride (responseOverride) {
    this._responseOverride = responseOverride
    return this
  }

  /**
   * Checks the data and builds the endpoint
   * @return {Endpoint} The final endpoint
   */
  build () {
    if (!this._name) {
      throw Error('You have to set a name to build an endpoint!')
    }

    if (!this._routerToUrlMapper) {
      throw Error('You have to set a url to build an endpoint!')
    }

    if (!this._mapper) {
      throw Error('You have to set a mapper to build an endpoint!')
    }

    if (!this._refetchLogic) {
      throw Error('You have to set a refetch logic to build an endpoint!')
    }

    return new Endpoint(this._name, this._routerToUrlMapper, this._mapper, this._refetchLogic, this._responseOverride)
  }
}

export default EndpointBuilder
