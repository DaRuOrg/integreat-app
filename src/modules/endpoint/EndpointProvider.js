import React from 'react'
import PropTypes from 'prop-types'
import Endpoint from './Endpoint'
import Error from '../common/containers/Error'

class EndpointProvider extends React.Component {
  static propTypes = {
    endpoints: PropTypes.arrayOf(PropTypes.instanceOf(Endpoint)).isRequired,
    children: PropTypes.element.isRequired
  }

  static childContextTypes = {
    getEndpoint: PropTypes.func.isRequired
  }

  getChildContext () {
    return {
      getEndpoint: (endpointName) => {
        const endpoint = this.endpoints[endpointName]
        if (!endpoint) {
          throw new Error('The endpoint ' + endpointName + ' was not found. Maybe you didn\'t wrap the ' +
            'withFetcher(...) call in a <EndpointProvider /> or the endpoint is not registered')
        }
        return endpoint
      }
    }
  }

  constructor (props, context) {
    super(props, context)
    this.endpoints = props.endpoints.reduce((accumulator, endpoint) => {
      accumulator[endpoint.stateName] = endpoint
      return accumulator
    }, {})
  }

  render () {
    return React.Children.only(this.props.children)
  }
}

export default EndpointProvider
