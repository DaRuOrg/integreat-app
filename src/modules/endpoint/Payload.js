/**
 * The payload gets stored in the redux store and holds the information about a etch
 */
class Payload {
  constructor (isFetching, data = null, error = null, requestUrl = null, fetchDate = null) {
    this._isFetching = isFetching
    this._fetchDate = fetchDate
    this._data = data
    this._error = error
    this._requestUrl = requestUrl
  }

  /**
   * @return {number} The date the fetch was initiated as serializable number
   */
  get fetchDate () {
    return this._fetchDate
  }

  /**
   * @return {boolean} If a fetch is going on
   */
  get isFetching () {
    return this._isFetching
  }

  /**
   * @return {*} The data which has been fetched or null
   */
  get data () {
    return this._data
  }

  /**
   * @return {boolean} If the {@link data} is ready to be used
   */
  ready () {
    return !!this._data
  }

  /**
   * @return {string} The error message if the fetch failed or null
   */
  get error () {
    return this._error
  }

  /**
   * @return {string} The url which was used to initiate the fetch
   */
  get requestUrl () {
    return this._requestUrl
  }
}

export default Payload
