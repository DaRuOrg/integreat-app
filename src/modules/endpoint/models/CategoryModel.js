// @flow

type categoryModel = {
  id: number,
  url: string,
  title: string,
  parentId: number,
  parentUrl: string,
  content: string,
  thumbnail: string,
  order: number,
  availableLanguages: any
}

class CategoryModel {
   _id: number
  _url: string
  _title: string
  _parentId: number
  _parentUrl: string
  _content: string
  _thumbnail: string
  _order: number
  _availableLanguages: any

  constructor (categoryModel: categoryModel) {
    this._id = categoryModel.id
    this._url = categoryModel.url
    this._title = categoryModel.title
    this._content = categoryModel.content
    this._parentId = categoryModel.parentId
    this._parentUrl = categoryModel.parentUrl
    this._thumbnail = categoryModel.thumbnail
    this._order = categoryModel.order
    this._availableLanguages = categoryModel.availableLanguages
  }

  get thumbnail (): string {
    return this._thumbnail
  }

  get id (): number {
    return this._id
  }

  get url (): string {
    return this._url
  }

  get title (): string {
    return this._title
  }

  get content (): string {
    return this._content
  }

  get parentId (): number {
    return this._parentId
  }

  get parentUrl (): string {
    return this._parentUrl
  }

  setParentUrl (parentUrl: string) {
    this._parentUrl = parentUrl
  }

  get order (): number {
    return this._order
  }

  get availableLanguages (): any {
    return this._availableLanguages
  }
}

export default CategoryModel
