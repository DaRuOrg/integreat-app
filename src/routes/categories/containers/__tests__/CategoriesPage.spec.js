import React from 'react'
import { mount, shallow } from 'enzyme'
import { Provider } from 'react-redux'
import configureMockStore from 'redux-mock-store'
import thunk from 'redux-thunk'

import ConnectedCategoriesPage, { CategoriesPage } from '../CategoriesPage'
import Payload from 'modules/endpoint/Payload'
import LocationModel from 'modules/endpoint/models/LocationModel'
import LanguageModel from 'modules/endpoint/models/LanguageModel'
import CategoryModel from 'modules/endpoint/models/CategoryModel'
import CategoriesMapModel from 'modules/endpoint/models/CategoriesMapModel'
import EndpointBuilder from 'modules/endpoint/EndpointBuilder'
import EndpointProvider from 'modules/endpoint/EndpointProvider'

describe('CategoriesPage', () => {
  const categoryModels = [
    new CategoryModel({
      id: 0,
      url: '/augsburg/de',
      title: 'augsburg'
    }),
    new CategoryModel({
      id: 3650,
      url: '/augsburg/de/anlaufstellen',
      title: 'Anlaufstellen zu sonstigen Themen',
      content: '',
      parentId: 0,
      parentUrl: '/augsburg/de',
      order: 75,
      availableLanguages: {
        en: 4361, ar: 4367, fa: 4368
      },
      thumbnail: 'https://cms.integreat-ap…/03/Hotline-150x150.png'
    }),
    new CategoryModel({
      id: 3649,
      url: '/augsburg/de/willkommen',
      title: 'Willkommen',
      content: '',
      parentId: 0,
      parentUrl: '/augsburg/de',
      order: 11,
      availableLanguages: {
        en: 4804, ar: 4819, fa: 4827
      },
      thumbnail: 'https://cms.integreat-ap…03/Beratung-150x150.png'
    }),
    new CategoryModel({
      id: 35,
      url: '/augsburg/de/willkommen/willkommen-in-augsburg',
      title: 'Willkommen in Augsburg',
      content: '<p>Willkommen in Augsbur…er Stadt Augsburg</p>\n',
      parentId: 3649,
      parentUrl: '/augsburg/de/willkommen',
      order: 1,
      availableLanguages: {
        en: '390',
        de: '711',
        ar: '397'
      },
      thumbnail: 'https://cms.integreat-ap…09/heart295-150x150.png'
    })
  ]

  const categories = new CategoriesMapModel(categoryModels)

  const locations = [
    new LocationModel({name: 'Augsburg', code: 'augsburg'}),
    new LocationModel({name: 'Stadt Regensburg', code: 'regensburg'}),
    new LocationModel({name: 'Werne', code: 'werne'})
  ]

  const location = 'augsburg'

  const languages = [
    new LanguageModel('en', 'English'),
    new LanguageModel('de', 'Deutsch'),
    new LanguageModel('ar', 'Arabic')
  ]

  const language = 'en'

  const categoriesEndpoint = new EndpointBuilder('categories')
    .withUrl('https://weird-endpoint/api.json')
    .withMapper(json => json)
    .withResponseOverride(categories)
    .build()

  const locationsEndpoint = new EndpointBuilder('locations')
    .withUrl('https://weird-endpoint/api.json')
    .withMapper(json => json)
    .withResponseOverride(locations)
    .build()

  const languagesEndpoint = new EndpointBuilder('languages')
    .withUrl('https://weird-endpoint/api.json')
    .withMapper(json => json)
    .withResponseOverride(languages)
    .build()

  test('should render a Page if page has no children', () => {
    const mockSetLanguageChangeUrls = jest.fn()
    const mockReplaceUrl = jest.fn()

    const wrapper = shallow(
      <CategoriesPage categories={categories}
                      locations={locations}
                      languages={languages}
                      location={location}
                      language={language}
                      path={'/augsburg/de/willkommen/willkommen-in-augsburg'}
                      setLanguageChangeUrls={mockSetLanguageChangeUrls}
                      replaceUrl={mockReplaceUrl} />
    )

    expect(wrapper).toMatchSnapshot()
  })

  test('should render a CategoryList if the path is neither root category nor a category with children', () => {
    const mockSetLanguageChangeUrls = jest.fn()
    const mockReplaceUrl = jest.fn()

    const wrapper = shallow(
      <CategoriesPage categories={categories}
                      locations={locations}
                      languages={languages}
                      location={location}
                      language={language}
                      path={'/augsburg/de/willkommen'}
                      setLanguageChangeUrls={mockSetLanguageChangeUrls}
                      replaceUrl={mockReplaceUrl} />
    )

    expect(wrapper).toMatchSnapshot()
  })

  test('should render CategoryTiles if the path is the root category', () => {
    const mockSetLanguageChangeUrls = jest.fn()
    const mockReplaceUrl = jest.fn()

    const wrapper = shallow(
      <CategoriesPage categories={categories}
                      locations={locations}
                      languages={languages}
                      location={location}
                      language={language}
                      path={'/augsburg/de'}
                      setLanguageChangeUrls={mockSetLanguageChangeUrls}
                      replaceUrl={mockReplaceUrl} />
    )

    expect(wrapper).toMatchSnapshot()
  })

  test('should render an Error if path is not valid', () => {
    const mockSetLanguageChangeUrls = jest.fn()
    const mockReplaceUrl = jest.fn()

    const wrapper = shallow(
      <CategoriesPage categories={categories}
                      locations={locations}
                      languages={languages}
                      location={location}
                      language={language}
                      path={'/augsburg/de/willkommen/willkommen-in-augsburg/test'}
                      setLanguageChangeUrls={mockSetLanguageChangeUrls}
                      replaceUrl={mockReplaceUrl} />
    )

    expect(wrapper).toMatchSnapshot()
  })

  test('should dispatch once in componentDidMount if the path is valid', () => {
    const mockSetLanguageChangeUrls = jest.fn()
    const mockReplaceUrl = jest.fn()

    const categoriesPage = shallow(
      <CategoriesPage categories={categories}
                      locations={locations}
                      languages={languages}
                      location={location}
                      language={language}
                      path={'/augsburg/de/willkommen'}
                      setLanguageChangeUrls={mockSetLanguageChangeUrls}
                      replaceUrl={mockReplaceUrl} />
    ).instance()

    expect(mockSetLanguageChangeUrls.mock.calls).toHaveLength(1)
    expect(mockSetLanguageChangeUrls).toBeCalledWith(
      categoriesPage.mapLanguageToUrl, languages, categoryModels[2].availableLanguages
    )
  })

  test('should not dispatch in componentDidMount if the path is invalid', () => {
    const mockSetLanguageChangeUrls = jest.fn()
    const mockReplaceUrl = jest.fn()

    shallow(
      <CategoriesPage categories={categories}
                      locations={locations}
                      languages={languages}
                      location={location}
                      language={language}
                      path={'/augsburg/de/willkommen/willkommen-in-augsburg/test'}
                      setLanguageChangeUrls={mockSetLanguageChangeUrls}
                      replaceUrl={mockReplaceUrl} />
    )

    expect(mockSetLanguageChangeUrls.mock.calls).toHaveLength(0)
  })

  test('should dispatch on prop update', () => {
    const mockSetLanguageChangeUrls = jest.fn()
    const mockReplaceUrl = jest.fn()

    const wrapper = shallow(
      <CategoriesPage categories={categories}
                      locations={locations}
                      languages={languages}
                      location={location}
                      language={language}
                      path={'/augsburg/de/willkommen'}
                      setLanguageChangeUrls={mockSetLanguageChangeUrls}
                      replaceUrl={mockReplaceUrl} />
    )

    const callCount = mockSetLanguageChangeUrls.mock.calls.length

    wrapper.setProps({...wrapper.props, path: '/augsburg/de/willkommen/willkommen-in-augsburg'})

    expect(mockSetLanguageChangeUrls.mock.calls).toHaveLength(callCount + 1)
    expect(mockSetLanguageChangeUrls).toBeCalledWith(
      wrapper.instance().mapLanguageToUrl, languages, categoryModels[3].availableLanguages
    )
  })

  test('should not dispatch on prop update if neither path nor categories are updated', () => {
    const mockSetLanguageChangeUrls = jest.fn()
    const mockReplaceUrl = jest.fn()

    const wrapper = shallow(
      <CategoriesPage categories={categories}
                      locations={locations}
                      languages={languages}
                      location={location}
                      language={language}
                      path={'/augsburg/de/willkommen'}
                      setLanguageChangeUrls={mockSetLanguageChangeUrls}
                      replaceUrl={mockReplaceUrl} />
    )

    const callCount = mockSetLanguageChangeUrls.mock.calls.length

    wrapper.setProps({...wrapper.props})

    expect(mockSetLanguageChangeUrls.mock.calls).toHaveLength(callCount)
  })

  test('mapLanguageToUrl', () => {
    const mockSetLanguageChangeUrls = jest.fn()
    const mockReplaceUrl = jest.fn()

    const mapLanguageToUrl = shallow(
      <CategoriesPage categories={categories}
                      locations={locations}
                      languages={languages}
                      location={location}
                      language={language}
                      path={'/augsburg/de/willkommen'}
                      setLanguageChangeUrls={mockSetLanguageChangeUrls}
                      replaceUrl={mockReplaceUrl} />
    ).instance().mapLanguageToUrl

    expect(mapLanguageToUrl('en', categoryModels[3].availableLanguages['en']))
      .toBe('/augsburg/en?id=' + categoryModels[3].availableLanguages['en'])
    expect(mapLanguageToUrl('en')).toBe('/augsburg/en')
  })

  test('getPdfFetchPath', () => {
    const mockSetLanguageChangeUrls = jest.fn()
    const mockReplaceUrl = jest.fn()

    const categoriesPage = shallow(
      <CategoriesPage categories={categories}
                      locations={locations}
                      languages={languages}
                      location={location}
                      language={language}
                      path={'/augsburg/de/willkommen'}
                      setLanguageChangeUrls={mockSetLanguageChangeUrls}
                      replaceUrl={mockReplaceUrl} />
    ).instance()

    expect(categoriesPage.getPdfFetchPath()).toBe('/augsburg/en/fetch-pdf?url=/augsburg/de/willkommen')
  })

  const mockStore = configureMockStore([thunk])

  describe('connect', () => {
    test('should map state to props', () => {
      const pathname = '/augsburg/de'
      const id = 1234
      const store = mockStore({
        categories: new Payload(false),
        languages: new Payload(false),
        locations: new Payload(false),
        router: {params: {location: location, language: language}, pathname: pathname, query: {id: id}}
      })

      const tree = mount(
        <Provider store={store}>
          <EndpointProvider endpoints={[categoriesEndpoint, locationsEndpoint, languagesEndpoint]}>
            <ConnectedCategoriesPage />
          </EndpointProvider>
        </Provider>
      )

      const categoriesPageProps = tree.find(ConnectedCategoriesPage).childAt(0).props()

      // todo add categories, locations and languages
      expect(categoriesPageProps).toEqual({
        location: location,
        language: language,
        path: pathname,
        categoryId: id,
        setLanguageChangeUrls: expect.any(Function),
        replaceUrl: expect.any(Function)
      })
    })

    // we don't have to test replaceUrl here, because it will be redundant after availableLanguages
    // in the cms are changed to the path instead of the id
    test('should map dispatch to props', () => {
      const pathname = '/augsburg/de'
      const store = mockStore({
        categories: new Payload(false),
        languages: new Payload(false),
        locations: new Payload(false),
        router: {params: {location: location, language: language}, pathname: pathname, query: {}}
      })

      const mapLanguageToUrl = (language, id) => 'test' + language + id

      const testUrls = {
        en: 'testenundefined',
        de: 'testde1235',
        ar: 'testar1236'
      }

      const availableLanguages = {
        de: '1235',
        ar: '1236'
      }

      const tree = mount(
        <Provider store={store}>
          <EndpointProvider endpoints={[categoriesEndpoint, locationsEndpoint, languagesEndpoint]}>
            <ConnectedCategoriesPage />
          </EndpointProvider>
        </Provider>
      )

      // todo expect setLanguageChangeUrls action to be in store, but as we don't get categories, locations and
      // languages from our mocked endpoint no action is dispatched

      const categoriesPageProps = tree.find(ConnectedCategoriesPage).childAt(0).props()

      let countActions = store.getActions().length

      categoriesPageProps.setLanguageChangeUrls(mapLanguageToUrl, languages, availableLanguages)
      expect(store.getActions()).toHaveLength(countActions + 1)

      expect(store.getActions()).toContainEqual({
        payload: testUrls,
        type: 'SET_LANGUAGE_CHANGE_URLS'
      })
    })
  })
})
