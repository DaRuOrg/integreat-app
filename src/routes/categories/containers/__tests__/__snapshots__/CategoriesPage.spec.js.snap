// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoriesPage should render CategoryTiles if the path is the root category 1`] = `
<div>
  <Breadcrumbs
    locations={
      Array [
        LocationModel {
          "_code": "augsburg",
          "_eventsEnabled": false,
          "_extrasEnabled": false,
          "_live": false,
          "_name": "Augsburg",
          "_sortCategory": "A",
          "_sortKey": "augsburg",
        },
        LocationModel {
          "_code": "regensburg",
          "_eventsEnabled": false,
          "_extrasEnabled": false,
          "_live": false,
          "_name": "Stadt Regensburg",
          "_sortCategory": "R",
          "_sortKey": "regensburg",
        },
        LocationModel {
          "_code": "werne",
          "_eventsEnabled": false,
          "_extrasEnabled": false,
          "_live": false,
          "_name": "Werne",
          "_sortCategory": "W",
          "_sortKey": "werne",
        },
      ]
    }
    parents={Array []}
  />
  <CategoryTiles
    categories={
      Array [
        CategoryModel {
          "_availableLanguages": Object {
            "ar": 4819,
            "en": 4804,
            "fa": 4827,
          },
          "_content": "",
          "_id": 3649,
          "_order": 11,
          "_parentId": 0,
          "_parentUrl": "/augsburg/de",
          "_thumbnail": "https://cms.integreat-ap…03/Beratung-150x150.png",
          "_title": "Willkommen",
          "_url": "/augsburg/de/willkommen",
        },
        CategoryModel {
          "_availableLanguages": Object {
            "ar": 4367,
            "en": 4361,
            "fa": 4368,
          },
          "_content": "",
          "_id": 3650,
          "_order": 75,
          "_parentId": 0,
          "_parentUrl": "/augsburg/de",
          "_thumbnail": "https://cms.integreat-ap…/03/Hotline-150x150.png",
          "_title": "Anlaufstellen zu sonstigen Themen",
          "_url": "/augsburg/de/anlaufstellen",
        },
      ]
    }
    locations={
      Array [
        LocationModel {
          "_code": "augsburg",
          "_eventsEnabled": false,
          "_extrasEnabled": false,
          "_live": false,
          "_name": "Augsburg",
          "_sortCategory": "A",
          "_sortKey": "augsburg",
        },
        LocationModel {
          "_code": "regensburg",
          "_eventsEnabled": false,
          "_extrasEnabled": false,
          "_live": false,
          "_name": "Stadt Regensburg",
          "_sortCategory": "R",
          "_sortKey": "regensburg",
        },
        LocationModel {
          "_code": "werne",
          "_eventsEnabled": false,
          "_extrasEnabled": false,
          "_live": false,
          "_name": "Werne",
          "_sortCategory": "W",
          "_sortKey": "werne",
        },
      ]
    }
    title="augsburg"
  />
  <PdfButton
    href="/augsburg/en/fetch-pdf?url=/augsburg/de"
  />
</div>
`;

exports[`CategoriesPage should render a CategoryList if the path is neither root category nor a category with children 1`] = `
<div>
  <Breadcrumbs
    locations={
      Array [
        LocationModel {
          "_code": "augsburg",
          "_eventsEnabled": false,
          "_extrasEnabled": false,
          "_live": false,
          "_name": "Augsburg",
          "_sortCategory": "A",
          "_sortKey": "augsburg",
        },
        LocationModel {
          "_code": "regensburg",
          "_eventsEnabled": false,
          "_extrasEnabled": false,
          "_live": false,
          "_name": "Stadt Regensburg",
          "_sortCategory": "R",
          "_sortKey": "regensburg",
        },
        LocationModel {
          "_code": "werne",
          "_eventsEnabled": false,
          "_extrasEnabled": false,
          "_live": false,
          "_name": "Werne",
          "_sortCategory": "W",
          "_sortKey": "werne",
        },
      ]
    }
    parents={
      Array [
        CategoryModel {
          "_availableLanguages": Object {},
          "_content": "",
          "_id": 0,
          "_order": 0,
          "_parentId": -1,
          "_parentUrl": null,
          "_thumbnail": null,
          "_title": "augsburg",
          "_url": "/augsburg/de",
        },
      ]
    }
  />
  <CategoryList
    categories={
      Array [
        CategoryModel {
          "_availableLanguages": Object {
            "ar": "397",
            "de": "711",
            "en": "390",
          },
          "_content": "<p>Willkommen in Augsbur…er Stadt Augsburg</p>
",
          "_id": 35,
          "_order": 1,
          "_parentId": 3649,
          "_parentUrl": "/augsburg/de/willkommen",
          "_thumbnail": "https://cms.integreat-ap…09/heart295-150x150.png",
          "_title": "Willkommen in Augsburg",
          "_url": "/augsburg/de/willkommen/willkommen-in-augsburg",
        },
      ]
    }
    content=""
    title="Willkommen"
  />
  <PdfButton
    href="/augsburg/en/fetch-pdf?url=/augsburg/de/willkommen"
  />
</div>
`;

exports[`CategoriesPage should render a Page if page has no children 1`] = `
<div>
  <Breadcrumbs
    locations={
      Array [
        LocationModel {
          "_code": "augsburg",
          "_eventsEnabled": false,
          "_extrasEnabled": false,
          "_live": false,
          "_name": "Augsburg",
          "_sortCategory": "A",
          "_sortKey": "augsburg",
        },
        LocationModel {
          "_code": "regensburg",
          "_eventsEnabled": false,
          "_extrasEnabled": false,
          "_live": false,
          "_name": "Stadt Regensburg",
          "_sortCategory": "R",
          "_sortKey": "regensburg",
        },
        LocationModel {
          "_code": "werne",
          "_eventsEnabled": false,
          "_extrasEnabled": false,
          "_live": false,
          "_name": "Werne",
          "_sortCategory": "W",
          "_sortKey": "werne",
        },
      ]
    }
    parents={
      Array [
        CategoryModel {
          "_availableLanguages": Object {},
          "_content": "",
          "_id": 0,
          "_order": 0,
          "_parentId": -1,
          "_parentUrl": null,
          "_thumbnail": null,
          "_title": "augsburg",
          "_url": "/augsburg/de",
        },
        CategoryModel {
          "_availableLanguages": Object {
            "ar": 4819,
            "en": 4804,
            "fa": 4827,
          },
          "_content": "",
          "_id": 3649,
          "_order": 11,
          "_parentId": 0,
          "_parentUrl": "/augsburg/de",
          "_thumbnail": "https://cms.integreat-ap…03/Beratung-150x150.png",
          "_title": "Willkommen",
          "_url": "/augsburg/de/willkommen",
        },
      ]
    }
  />
  <Page
    content="<p>Willkommen in Augsbur…er Stadt Augsburg</p>
"
    title="Willkommen in Augsburg"
  />
  <PdfButton
    href="/augsburg/en/fetch-pdf?url=/augsburg/de/willkommen/willkommen-in-augsburg"
  />
</div>
`;

exports[`CategoriesPage should render an Error if path is not valid 1`] = `
<Translate(Failure)
  error="not-found:page.notFound"
/>
`;
